tck/src/main/java/org/reactivestreams/tck/TestEnvironment.java

```diff
@@ -973,6 +977,21 @@ public void expectCompletion(long timeoutMillis, String errorMsg) throws Interru
       } // else, ok
     }
 
+    public boolean tryExpectCompletion(long timeoutMillis) throws InterruptedException {
+      while (timeoutMillis-- > 0) {
+        Optional<T> value = abq.peek();
+        if (value != null) {
+          if (!value.isDefined()) {
+            abq.poll();
+            return true;
+          }
+          return false;
+        }
+        Thread.sleep(1);
```

Comment:
1) There is no such method of `abq.poll(arg0)`: [Javadoc](https://docs.oracle.com/javase/7/docs/api/java/util/concurrent/ArrayBlockingQueue.html)
2) The purpose of the method is to consume a non-defined, non-null `Optional` and return true indicating the next event in line was a terminal event. If the next event is not a terminal event, don't consume it and return false and leave up to the test what to do with that situation. In other terms, `poll()` is not good because it consumes the item unconditionally.